plugin/pkg/admission/serviceaccount/admission.go

```diff
@@ -284,6 +294,10 @@ func (s *serviceAccount) getReferencedServiceAccountToken(serviceAccount *api.Se
 		}
 	}
 
+	if errIfTokenNotFound {
+		return "", fmt.Errorf("no secret found which is a service acount token for service account  %s/%s, secrets: %v, service account tokens: %v",
```

Comment:
but it loses all the detail the message on line 350 adds about what you actually need to do